src/prefect/artifacts.py

```diff
@@ -0,0 +1,201 @@
+        artifact_args["flow_run_id"] = task_run_ctx.task_run.flow_run_id
+    elif flow_run_ctx:
+        artifact_args["flow_run_id"] = flow_run_ctx.flow_run.id
+
+    if key is not None:
+        artifact_args["key"] = key
+    if type is not None:
+        artifact_args["type"] = type
+    if description is not None:
+        artifact_args["description"] = description
+    if data is not None:
+        artifact_args["data"] = data
+    if metadata is not None:
+        artifact_args["metadata_"] = metadata
+
+    artifact = ArtifactCreate(**artifact_args)
+
+    return await client.create_artifact(artifact=artifact)
+
+
+@sync_compatible
+async def create_link(
+    link: str,
+    link_text: Optional[str] = None,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    metadata: Optional[Dict[str, Any]] = None,
+) -> UUID:
+    """
+    Create a link artifact.
+
+    Args:
+        - link: The link to create.
+
+    Returns:
+        - The table artifact ID.
+    """
+    formatted_link = f"[{link_text}]({link})" if link_text else f"[{link}]({link})"
+    artifact = await _create_artifact(
+        key=name,
+        type="markdown",
+        description=description,
+        data=formatted_link,
+        metadata=metadata,
+    )
+
+    return artifact.id
+
+
+@sync_compatible
+async def create_markdown(
+    markdown: str,
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    metadata: Optional[Dict[str, Any]] = None,
+) -> UUID:
+    """
+    Create a markdown artifact.
+
+    Args:
+        - markdown: The markdown to create.
+
+    Returns:
+        - The table artifact ID.
+    """
+    artifact = await _create_artifact(
+        key=name,
+        type="markdown",
+        description=description,
+        data=markdown,
+        metadata=metadata,
+    )
+
+    return artifact.id
+
+
+@sync_compatible
+async def create_table(
+    table: Union[Dict[str, List[Any]], List[Dict[str, Any]]],
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    metadata: Optional[Dict[str, Any]] = None,
+) -> UUID:
+    """
+    Create a table artifact.
+
+    Args:
+        - table: The table to create.
+
+    Returns:
+        - The table artifact ID.
+    """
+    formatted_table = json.dumps(table) if isinstance(table, list) else table
+
+    artifact = await _create_artifact(
+        key=name,
+        type="table",
+        description=description,
+        data=formatted_table,
+        metadata=metadata,
+    )
+
+    return artifact.id
+
+
+@inject_client
+async def _read_artifact(
+    artifact_id: Union[str, UUID],
+    client: Optional[PrefectClient] = None,
+) -> Artifact:
+    """
+    Helper function to read an artifact.
+    """
+    if isinstance(artifact_id, str):
+        artifact_id = UUID(artifact_id)
+
+    return await client.read_artifact(artifact_id=artifact_id)
+
+
+@sync_compatible
+async def read_markdown(
+    artifact_id: Union[str, UUID],
```

Comment:
Shouldn't users be fetching artifacts by key not id?

Reply:
It probably doesn't make sense to expose different `read_` methods. We can parse the artifact based on the type that we read from the API.

Reply:
There will be many artifacts with the same key, so we could always return the latest artifact on that key? It seems less specific. There could be multiple ways to retrieve artifacts?

Reply:
I believe we would need a new model method "read_artifact_by_key" which would require adding the ArtifactCollection table which doesn't exist in oss in order to to optimally retrieve the (key, id) we want. Perhaps this PR should just include creating artifacts, and a follow-up PR would get oss on par with nebula before a third PR would add the read interface.

Reply:
We don't need to have retrieval be the optimized version to add it here.

Reply:
I'm agnostic to the order in which the functionality is merged, but I agree with @madkinsz that the key should be the primary way to retrieve the artifact.